components/engine_rocks/src/raft_engine.rs

```diff
@@ -233,6 +257,13 @@ impl RaftLogBatch for RocksWriteBatch {
     }
 
     fn cut_logs(&mut self, raft_group_id: u64, from: u64, to: u64) {
+        {
+            let mut ctx = RAFT_LOG_GC_CONTEXT.write().unwrap();
+            if ctx.gc_on_compaction {
+                ctx.apply_idxs.insert(raft_group_id, to);
```

Comment:
`cut_logs` just delete the entries in the write batch, not the db. Seems shouldn't update gc context here.